File: frontend/src/types/inventory.ts
export type InventoryStatus =
  "Warehouse" | "Issued" | "Returned" | "Written Off";
export type OwnershipType = "Own" | "State";

export interface InventoryItem {
  id: number;
  inventory_number: string;
  invoice_number: string;
  invoice_date: string;
  item_name: string;
  nomenclature_code: string | null;
  serial_number: string | null;
  unit: string;
  category: string;
  quantity: number;
  price: number;
  total_price: number;
  issued_to: string | null;
  location: string | null;
  issued_date: string | null;
  ownership: string | null;
  department: string | null;
  invoice_receiver: string | null;
  status: InventoryStatus;
  note: string | null;
  created_at: string;
  updated_at: string;
  days_issued: number | null;
}

export interface InventoryListResponse {
  items: InventoryItem[];
  total: number;
  page: number;
  limit: number;
}

export interface Statistics {
  total_items: number;
  warehouse_items: number;
  issued_items: number;
  returned_items: number;
  written_off_items: number;
  total_value: number;
}

export interface ApiResponse<T> {
  success: boolean;
  message: string;
  data: T;
}

export interface InventoryFilters {
  page: number;
  limit: number;
  search: string;
  status: string;
  category: string;
  department: string;
  ownership: string;
  sort: string;
  order: "asc" | "desc";
}

File: backend/app/schemas/inventory.py
from datetime import date, datetime
from pydantic import BaseModel, Field


class InventoryCreate(BaseModel):
    invoice_number: str
    invoice_date: date
    item_name: str
    nomenclature_code: str | None = None
    serial_number: str | None = None
    unit: str
    category: str
    quantity: int = Field(default=1, ge=1)
    price: float = Field(default=0.0, ge=0)
    total_price: float = Field(default=0.0, ge=0)
    issued_to: str | None = None
    location: str | None = None
    issued_date: date | None = None
    ownership: str | None = None
    department: str | None = None
    invoice_receiver: str | None = None
    status: str = "Warehouse"
    note: str | None = None


class InventoryUpdate(BaseModel):
    invoice_number: str | None = None
    invoice_date: date | None = None
    item_name: str | None = None
    nomenclature_code: str | None = None
    serial_number: str | None = None
    unit: str | None = None
    category: str | None = None
    quantity: int | None = Field(default=None, ge=1)
    price: float | None = Field(default=None, ge=0)
    total_price: float | None = Field(default=None, ge=0)
    issued_to: str | None = None
    location: str | None = None
    issued_date: date | None = None
    ownership: str | None = None
    department: str | None = None
    invoice_receiver: str | None = None
    status: str | None = None
    note: str | None = None


class InventoryOut(BaseModel):
    id: int
    inventory_number: str
    invoice_number: str
    invoice_date: date
    item_name: str
    nomenclature_code: str | None
    serial_number: str | None
    unit: str
    category: str
    quantity: int
    price: float
    total_price: float
    issued_to: str | None
    location: str | None
    issued_date: date | None
    ownership: str | None
    department: str | None
    invoice_receiver: str | None
    status: str
    note: str | None
    created_at: datetime
    updated_at: datetime
    days_issued: int | None = None

    model_config = {"from_attributes": True}


class InventoryListOut(BaseModel):
    items: list[InventoryOut]
    total: int
    page: int
    limit: int
